StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/OrderingRules/SA1206CSharp8UnitTests.cs

```diff
@@ -3,9 +3,70 @@
 
 namespace StyleCop.Analyzers.Test.CSharp8.OrderingRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
     using StyleCop.Analyzers.Test.CSharp7.OrderingRules;
+    using Xunit;
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.OrderingRules.SA1206DeclarationKeywordsMustFollowOrder,
+        StyleCop.Analyzers.OrderingRules.SA1206CodeFixProvider>;
 
     public partial class SA1206CSharp8UnitTests : SA1206CSharp7UnitTests
     {
+        [Fact]
+        [WorkItem(3005, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3005")]
+        public async Task TestStaticLocalFunctionIncorrectOrderAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    public void Method()
+    {
+        async {|#0:static|} void Local()
+        {
+        }
+    }
+}";
+            var fixedCode = @"public class TestClass
+{
+    public void Method()
+    {
+        static async void Local()
+        {
+        }
+    }
+}";
+
+            var expected = Diagnostic().WithArguments("static", "async").WithLocation(0);
+
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        [WorkItem(3005, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3005")]
+        public async Task TestStaticUnsafeLocalFunctionIncorrectOrderAsync()
+        {
+            var testCode = @"public class TestClass
```

Comment:
Have you considered using newer syntax for these new tests that adds syntax coloring and avoids any escaping requirements? e.g.

```cs
           var testCode = /* lang=c#-test */ """
               public class TestClass
               {
                 // ...
               }
               """;
```

Reply:
Yes, will convert over to that syntax after I finish work on the current language feature updates.